pkgs/desktops/gnome/misc/gnome-flashback/default.nix

```diff
@@ -155,15 +156,38 @@ let
             '';
           };
 
-          executable = writeShellScriptBin "gnome-flashback-${wmName}" ''
-            if [ -z $XDG_CURRENT_DESKTOP ]; then
-              export XDG_CURRENT_DESKTOP="GNOME-Flashback:GNOME"
-            fi
-
-            export XDG_DATA_DIRS=${wmApplication}/share:${gnomeSession}/share:${gnome-flashback}/share:${gnome-panel}/share:$XDG_DATA_DIRS
+          # gnome-panel will only look for applets in a single directory so symlink them into here.
+          panelModulesEnv = buildEnv {
+            name = "panel-modules-env";
+            # We always want to find the built-in panel applets.
+            paths = [ gnome-panel gnome-flashback ] ++ panelModulePackages;
+            pathsToLink = [ "/lib/gnome-panel/modules" ];
+          };
 
-            exec ${gnome-session}/bin/gnome-session --session=gnome-flashback-${wmName} "$@"
-          '';
+          executable = stdenv.mkDerivation {
+            name = "gnome-flashback-${wmName}";
+            nativeBuildInputs = [ wrapGAppsHook ];
+            buildInputs = [ glib gnome-panel gnome-flashback ] ++ panelModulePackages;
```

Comment:
glib for setup hook should go to wrapGAppsHook

Reply:
If you mean it's not necessary, I took it out.

Reply:
Sorry, meant `nativeBuildInputs`. Should probably take a nap. Though it might actually be propagated, not sure.

Reply:
Done.